src/pages/index.js

```diff
@@ -20,88 +64,191 @@ const cardSection = new Section(
+  cards.forEach((cardData) => {
+    const cardElement = createCard(cardData); // Generate card using createCard
+    cardSection.addItem(cardElement); // Properly add to `.gallery__cards`
+    const likeBtn = document.querySelector('.card__like-button');
+    console.log(cardData.isLiked)
+    if(cardData.isLiked){
+      console.log(cardData._id);  
+      likeBtn.classList.toggle('card__like-button--active');
+    }else{
+      likeBtn.classList.remove('card__like-button--active');
+    }
+  });
+});
 
-//---------------------FORMS AND SUBMITS--------------------------
+//---------------------FORMS AND VALIDATION AND SUBMIT BUTTONS RESPONSES--------------------------
 
 // Initialize form validation
 const addCardValidator = new FormValidator(config, document.querySelector("#add__modal_form"));
 const profileValidator = new FormValidator(config, document.querySelector("#edit__modal_form"));
+const avatarValidator = new FormValidator(config, document.querySelector("#avatar__modal_form"));
 addCardValidator.enableValidation();
 profileValidator.enableValidation();
+avatarValidator.enableValidation();
 
 //-----------------------POPUPS--------------------------------
-
 // Create Popups
 const imagePopup = new PopupWithImage("#modal__picture");
 imagePopup.setEventListeners();
 
 //  Initialize `UserInfo`
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__description"
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar"
 });
 
+function submitButtonTextChanger(button){
+  return button.textContent = 'Saving...'
+}
+
 const profilePopup = new PopupWithForm("#profileModal", (formData) => {
-  userInfo.setUserInfo(formData);
+  const originalText = profileSubmitButton.textContent;
+  
+  api.updateUserInfo(formData.title, formData.about) // sends to Api.js to save to database
+    .then((updatedData) => {
+      submitButtonTextChanger(profileSubmitButton);
+      console.log('Changed profile submit button to saving...')
+      userInfo.setUserInfo(formData);
+      profilePopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(()=>{
+      profileSubmitButton.textContent = originalText;
+      console.log('changing prof submit button text back to original')
+    })
 });
+
 profilePopup.setEventListeners();
 
+
 const addCardPopup = new PopupWithForm("#modal__add", (formData) => {
+const originalText = addCardPopupSubmit.textContent;
+
   const newCardData = {
     name: formData.name.trim(),
     link: formData.url.trim(),
   };
 
   if (newCardData.name && newCardData.link) {
-    cardSection.addItem(createCard(newCardData));
-
-    // Only reset form on submit
-    document.querySelector("#add__modal_form").reset();
-
-    // Close popup after adding a card
-    addCardPopup.close();
+    api.updateNewCards(newCardData) 
+      .then((savedCard) => {
+        submitButtonTextChanger(addCardPopupSubmit);
```

Comment:
`submitButtonTextChanger(addCardPopupSubmit);`  starts the loading effect, so it should be **before** the request